crates/typed-store/src/rocks/iter.rs

```diff
@@ -54,6 +54,11 @@ impl<'a, K: DeserializeOwned, V: DeserializeOwned> Iterator for Iter<'a, K, V> {
                 .db_iter
                 .value()
                 .expect("Valid iterator failed to get value");
+            self.db_metrics
```

Comment:
I think we do want to differentiate between get and iter. Maybe take the rocksdb_iter_bytes out of the sampling condition?